File: src/test/java/ru/romanow/jpa/web/config/DatabaseTestConfiguration.java
package ru.romanow.jpa.web.config;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.springframework.context.annotation.Bean;
import org.testcontainers.containers.PostgreSQLContainer;
import org.testcontainers.containers.output.Slf4jLogConsumer;

@TestConfiguration
public class DatabaseTestConfiguration {
    private static final Logger logger = LoggerFactory.getLogger(DatabaseTestConfiguration.class);

    private static final String POSTGRES_IMAGE = "postgres:15-alpine";
    private static final String DATABASE_NAME = "example";
    private static final String USERNAME = "program";
    private static final String PASSWORD = "test";

    @Bean
    @ServiceConnection
    public PostgreSQLContainer<?> postgres() {
        return new PostgreSQLContainer<>(POSTGRES_IMAGE)
            .withUsername(USERNAME)
            .withPassword(PASSWORD)
            .withDatabaseName(DATABASE_NAME)
            .withLogConsumer(new Slf4jLogConsumer(logger));
    }
}

File: src/test/java/ru/romanow/jpa/config/DatabaseTestConfiguration.kt
package ru.romanow.jpa.config

import org.springframework.boot.test.context.TestConfiguration
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.springframework.context.annotation.Bean
import org.testcontainers.containers.PostgreSQLContainer

typealias CustomPostgresContainer = PostgreSQLContainer<*>

@TestConfiguration
class DatabaseTestConfiguration {
    @Bean
    @ServiceConnection
    fun postgres(): PostgreSQLContainer<*> {
        return CustomPostgresContainer(POSTGRES_IMAGE)
            .withUsername(USERNAME)
            .withPassword(PASSWORD)
            .withDatabaseName(DATABASE_NAME)
    }

    companion object {
        private const val POSTGRES_IMAGE = "postgres:15-alpine"
        private const val DATABASE_NAME = "example"
        private const val USERNAME = "program"
        private const val PASSWORD = "test"
    }
}
